### PR TITLE
Fixing the custom looper so it's scoped to triggers, fixing logging i…

### PR DESCRIPTION
**Pull request by StabbyCutyou**
…ssues with the http handlers

ping @rossnanop @jandre 

Would be good to fix this asap :)

### CODE REVIEW

File: templates/server/http/handler_trigger_x.template [hunk 1]
```diff
@@ -73,7 +73,7 @@ func (h *{{ .Name }}TriggerHandler) ServeHTTP(w http.ResponseWriter, r *http.Req
 	// note to anyone reading this: Do not go adding in sync.Pool without understanding how they
 	// work and how to profile to see if adding one will actually make a difference :) It's tricky iirc.
 	// Thats why this is a TODO and not a TODONE
-	l := plog.NewBufferedLogger()
+	l := plog.NewBufferedLogger(plog.Info)
```
Comment (StabbyCutyou) [inline templates/server/http/handler_trigger_x.template]: Triggers are always test, so they always go info level

File: templates/server/http/handler_x.template [hunk 2]
```diff
@@ -96,7 +100,8 @@ func (h *{{ .Name }}Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 
 	action := actions.{{ .Name }}Action{}
 	output := &actions.{{ .Name }}Output{}
-	if strings.HasSuffix(r.URL.String(), "test") {
+	if isTest {
+		l.SetLevel(plog.Info)
```
Comment (rossnanop) [inline templates/server/http/handler_x.template]: It looks like you set this level to Info twice in this method? Do you only need to do this once?

File: templates/triggers/trigger_x.template [hunk 3]
```diff
@@ -10,6 +10,11 @@ import (
     "{{ .PackageRoot }}/connection"
 )
 
+// CustomLooper is an interface you can implement to override RunLoop behavior
+type {{ .Name}}TriggerCustomLooper interface {
```
Comment (rossnanop) [inline templates/triggers/trigger_x.template]: Can you update the comment as well to have the {{ .Name }}? And add a space after .Name for style.

File: templates/server/http/handler_x.template [hunk 4]
```diff
@@ -70,7 +70,11 @@ func (h *{{ .Name }}Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	// note to anyone reading this: Do not go adding in sync.Pool without understanding how they
 	// work and how to profile to see if adding one will actually make a difference :) It's tricky iirc.
 	// Thats why this is a TODO and not a TODONE
+	isTest := strings.HasSuffix(r.URL.String(), "test")
 	l := plog.NewBufferedLogger(plog.Error)
+	if isTest {
+		l.SetLevel(plog.Info)
```
Comment (rossnanop) [inline templates/server/http/handler_x.template]: See below about setting twice.

File: templates/cmd/main.template [hunk 5]
```diff
@@ -256,7 +256,7 @@ func runTriggerDaemon(ctx context.Context, mode string, b *message.BodyV1, isDeb
 		// Start the read loop
 		go t.ReadLoop(ctx, l)
 		var ti interface{} = t
-		if cl, ok := ti.(triggers.CustomLooper); ok {
+		if cl, ok := ti.(triggers.{{ $p.Name }}TriggerCustomLooper); ok {
```
Comment (rossnanop) [inline templates/cmd/main.template]: Is there only ever one main.go in /cmd? Isn't runTriggerDaemon a generic runner for all triggers? Does it make sense for this to be the specific named interface? Sorry for the flurry of questions.  Debating if this needs to be rethought.

Comment (StabbyCutyou) [inline templates/cmd/main.template]: So, the reason this had to change was:

You had the original interface definition in trigger_template, which is generated once per trigger.

So, if there are 3 triggers, it gets declared 3 times. Plugins won't compile.

But the override for Run is a method which returns a type specific to a plugin. Therefore, the interface itself cannot be generic for all triggers, it has to be 1 per trigger like everything else.

Because of this, in the loop we're in here (which is, one per trigger), we need to get the right interface for the type of trigger output we expect per this trigger.

Which is as easy as setting the name.

That said, if there were a way that interface could be genericized into a single one, that might be better, but i don't love giving up the type safety of the current design. It's all generated code anyways, so long as the result behaves the same we should be fine to change it (although plugins may need to change if they update/regenerate).

File: templates/server/http/handler_x.template (hunk 4 above)
Comment (StabbyCutyou) [inline templates/server/http/handler_x.template]: Fixed

File: templates/cmd/main.template (hunk 5 above)
Comment (StabbyCutyou) [inline templates/cmd/main.template]: Er, sorry - it doesn't return a custom type to the trigger, it accepts in a custom type.

Everything else is the same. If we make that input an interface{} it puts more work on the author when they fill out the iface method - maybe that's ok VS an interface for each type of trigger, but seems to be a QoL downgrade since once you're in the context of adding a custom loop to a trigger, you will never have any input other than the correct type.

In other words, the additional interfaces are all behind the scenes boilerplate, not something the user touches, thus i'm ok with that part being ugly in the face of a simpler ux.

**Comment by rossnanop** [comment]
@StabbyCutyou LGTM
